velox/vector/arrow/Bridge.cpp

```diff
@@ -1518,8 +1518,18 @@ VectorPtr importFromArrowImpl(
         arrowArray.n_buffers,
         2,
         "Primitive types expect two buffers as input.");
-    auto values = wrapInBufferView(
-        arrowArray.buffers[1], arrowArray.length * type->cppSizeInBytes());
+    BufferPtr values = nullptr;
+    if (type->isShortDecimal()) {
```

Comment:
Maybe it's better to create a new branch before `type->isPrimitiveType()`, and add a new method `createShortDecimalVector` for the special handling.